src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,70 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
```

Comment:
Consider using `Map.of()` to initialize the `instances` map with predefined mappings of interfaces to their implementations. This will make the code more robust and prevent potential runtime exceptions when an interface is not mapped.